`chbder_vector` not creating the correct results

The function `cspyce.chbder` returns the value of a polynomial and its first `nderiv` derivatives, evaluated at the input `x`, using the coefficients of the Chebyshev expansion of the polynomial. Here are the inputs and returns for the vectorized version:


```
    Inputs:
      float[_,*] cp     = degp+1 Chebyshev polynomial coefficients.
      float[_,2] x2s    = Transformation parameters of polynomial.
      float[_]   x      = Value for which the polynomial is to be evaluated.
      int        nderiv = The number of derivatives to compute.
    
    Returns:
      float[_]   dpdxs  = Array of the derivatives of the polynomial.
```


It should be noted that, if the arguments used in the vectorized version are the same as the scalar version, it should give the same answer. Here is the scalar version of the test:
```
import cspyce as cs
import numpy.testing as npt

cp = [1.0, 3.0, 0.5, 1.0, 0.5, -1.0, 1.0]
x2s = [0.5, 3.0]
dpdxs = cs.chbder(cp, x2s, 1.0, 3)
npt.assert_array_almost_equal([-0.340878, 0.382716, 4.288066, -1.514403],
                              dpdxs)
```

This passes no problem. But when this same test is vectorized (and nothing else changes), it gives an error:

```
AssertionError: 
Arrays are not almost equal to 6 decimals

Mismatched elements: 4 / 4 (100%)
Max absolute difference: 4.288066
Max relative difference: 1.340878
 x: array([-0.340878,  0.382716,  4.288066, -1.514403])
 y: array([1., 0., 0., 0.])
```

It is likely that there is a bug for `chbder_vector` that needs to be addressed.

--- a/swig/cspyce0_part2.i
+++ b/swig/cspyce0_part2.i
@@ -491,7 +491,7 @@ extern SpiceInt bsrchi_c(
         // out21 is dpdxs
 
         int nderiv_plus_1 = nderiv + 1;
-        int degp = in21_dim1 - 1;
+        int degp = in21_dim2 - 1;
 
         *out21 = NULL;
         *out21_dim1 = 0;

